modular_citadel/code/modules/mob/living/silicon/robot/robot_movement.dm

```diff
@@ -4,8 +4,8 @@
 /mob/living/silicon/robot/Move(NewLoc, direct)
 	. = ..()
 	if(. && sprinting && !(movement_type & FLYING) && canmove && !resting)
-		if(istype(cell))
-			cell.charge -= 25
+		if(!(cell?.use(25)))
+			togglesprint()
```

Comment:
Make sure to use the arg to force disable sprinting. It'd also probably be good audio feedback to play the sprint deactivate sound here, maybe mix it in with light_flicker.ogg and slap a vocoder filter or phaser effect on top of it.

Reply:
`togglesprint()` has no arg and already plays the sprint toggle sound.
But, fine.